pkg/agent/core/ngt/service/ngt.go

```diff
@@ -427,13 +351,92 @@ func (n *ngt) initNGT(opts ...core.Option) (err error) {
 	return nil
 }
 
+func (n *ngt) initNGT(opts ...core.Option) (err error) {
+	if n.kvs == nil {
+		n.kvs = kvs.New(kvs.WithConcurrency(n.kvsdbConcurrency))
+	}
+	if n.inMem {
+		log.Debug("vald agent starts with in-memory mode")
+		n.core, err = core.New(opts...)
+		return err
+	}
+
+	ctx := context.Background()
+	err = n.load(ctx, n.path, opts...)
+	var current uint64
+	if err != nil {
+		if !n.enableCopyOnWrite {
+			log.Debug("failed to load vald index from %s\t error: %v", n.path, err)
+			if n.kvs == nil {
+				n.kvs = kvs.New(kvs.WithConcurrency(n.kvsdbConcurrency))
+			}
+			n.core, err = core.New(append(opts, core.WithIndexPath(n.path))...)
+			return err
+		}
+		if errors.Is(err, errors.ErrIndicesAreTooFewComparedToMetadata) && n.kvs != nil {
+			current = n.kvs.Len()
+			log.Warnf(
+				"load vald primary index success from %s\t error: %v\tbut index data are too few %d compared to metadata count now trying to load from old copied index data from %s and compare them",
+				n.path,
+				err,
+				current,
+				n.oldPath,
+			)
+		} else {
+			log.Warnf("failed to load vald primary index from %s\t error: %v\ttrying to load from old copied index data from %s", n.path, err, n.oldPath)
+		}
+	}
+	err = n.load(ctx, n.oldPath, opts...)
+	if err == nil {
+		if current != 0 && n.kvs.Len() < current {
+			log.Warnf(
+				"load vald secondary index success from %s\t error: %v\tbut index data are too few %d compared to primary data now trying to load from primary index data again from %s and start up with them",
+				n.oldPath,
+				err,
+				n.kvs.Len(),
+				n.oldPath,
+			)
+			err = n.load(ctx, n.path, opts...)
+			if err == nil {
+				return nil
+			}
```

Comment:
I suppose, if err is not nil, should we output another log in the below line?
There is no chance to output this err in current implementation.

Reply:
it would be shown at L407

Reply:
my eyes were 🐸